Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,251 @@
+
+function fail_test() {
+    LogErr "$*"
+    SetTestStateFailed
+    exit 0
+}
+
+function verify_directory_exists() {
+    failed_assertions=false
+    for dir in "${dir_list[@]}"
+    do
+        if [ -d "$dir" ]; then
+            LogErr "Directory $dir exists."
+            failed_assertions=true
+        fi
+    done
+
+    if $failed_assertions; then
+        LogErr "Some/All directories exists"
+        echo "$failed_assertions"
+    fi
+}
+
+function verify_distro() {
+    case $DISTRO in
+        redhat* | centos* | suse* | debian* | ubuntu*)
+            LogMsg "Supported Distro family: $DISTRO"
+            ;;
+        *)
+            fail_test "Unsupported Distro family: $DISTRO"
+            ;;
+    esac
+}
+
+function verify_expected_file() {
+    if [ ! -f "$1" ]; then
+        fail_test "${1} doesn't exist."
+    fi
+}
+
+function verify_da_pid() {
+    [ ! -f $da_pid_file ] && fail_test "$da_pid_file does not exist."
+
+    if x=$(sed -n '/^[1-9][0-9]*$/p;q' $da_pid_file); then
+        if strings /proc/$x/cmdline | fgrep -q  -e "microsoft-dependency-agent-manager"; then
+            LogMsg "PID matched"
+        else
+            fail_test "PID not matched"
+        fi
+    else
+        fail_test "PID not found in DA_PID"
+    fi
+}
+
+function check_prereqs() {
+    LogMsg "Checking for preconditions"
+
+    if [[ $EUID -ne 0 ]]; then
+        fail_test "This script must be run as root"
+    fi
+
+    verify_distro
+
+    directory_exists=$(verify_directory_exists)
+    if [ "directory_exists" == "true" ]; then
+        SetTestStateAborted
+        exit 0
+    fi
+
+    LogMsg "Preconditions met"
+}
+
+function download_da_linux_installer() {
+    LogMsg "Download Dependency Agent Linux installer"
+
+    if ! curl -L https://aka.ms/dependencyagentlinux -o InstallDependencyAgent-Linux64.bin; then
+        fail_test Download failed
+    fi
+
+    chmod +x InstallDependencyAgent-Linux64.bin
+}
+
+function verify_install_da() {
+    LogMsg "Starting Install tests"
+
+    ./InstallDependencyAgent-Linux64.bin -vme
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        fail_test "Install failed with exit code ${ret}"
+    fi
+    LogMsg "Dependency Agent installed successfully"
+
+    verify_expected_file "$uninstaller"
+    verify_expected_file "/etc/init.d/microsoft-dependency-agent"
+    verify_expected_file "/var/opt/microsoft/dependency-agent/log/install.log"
+
+    bin_version=$(./InstallDependencyAgent-Linux64.bin --version | awk '{print $5}')
+    install_log_version=$(sed -n 's/^Dependency Agent version\.revision: //p' /var/opt/microsoft/dependency-agent/log/install.log)
+    if [ "$bin_version" -ne "$install_log_version" ]; then
+        fail_test "Version mismatch between bin version($bin_version) and install log version($install_log_version)"
+    fi
+    LogMsg "Version matches between bin version and install log version"
+
+    LogMsg "Install tests passed successfully"
```

Comment:
"passed" and "successfully" are redundant.

Reply:
Done